src/main/java/com/iexec/core/task/update/TaskUpdate.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.task.update;
+
+import com.iexec.core.task.Task;
+import com.iexec.core.task.TaskStatus;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Comparator;
+import java.util.Date;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentMap;
+
+@Slf4j
+class TaskUpdate implements Runnable, Comparable<TaskUpdate> {
+    private final Task task;
+    private final ConcurrentMap<String, Object> locks;
+    private final TaskUpdateRequestConsumer consumer;
+
+    TaskUpdate(Task task,
+               ConcurrentMap<String, Object> locks,
+               TaskUpdateRequestConsumer consumer) {
+        this.task = task;
+        this.locks = locks;
+        this.consumer = consumer;
+    }
+
+    public Task getTask() {
+        return task;
+    }
+
+    public String getChainTaskId() {
+        return task.getChainTaskId();
+    }
+
+    private TaskStatus getCurrentStatus() {
+        return task.getCurrentStatus();
+    }
+
+    private Date getContributionDeadline() {
+        return task.getContributionDeadline();
+    }
+
+    /**
+     * Updates a task.
+     * <br>
+     * 2 updates can be run in parallel if they don't target the same task.
+     * Otherwise, the second update will wait until the first one is achieved.
+     */
+    @Override
+    public void run() {
```

Comment:
As the `Runnable` interface is directly implemented, the `@Override` annotation is unnecessary here.
We implement the method and don't rely on any previous implementation.

Reply:
I don't get your point. `@Override` tells the compiler to check the method does effectively override or implement a method declared in a supertype. That's some security that should, IMHO, always be present.

Reply:
If you implement an interface without defining implementations for its methods, the compilation will fail.
I don't see the added security of the `@Override` annotation in this case, unless it helps to quickly detect inherited or implemented methods in the `TaskUpdate` class.

Reply:
I see it the other way: it enforces the class to implement the `Runnable` interface. It's handy if we ever refactor it and miss this `implements Runnable` declaration.

Reply:
I understand your point, thanks